### PR TITLE
Adding ability to publish images to the azure portal.

### PR DESCRIPTION
**Pull request by kwoodson**
The image publishing scripts for Azure.

### CODE REVIEW

File: playbooks/azure/openshift-cluster/create_offer.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      az storage container generate-sas --name {{ openshift_azure_container }} --account-name {{ openshift_azure_storage_account }} --permissions rl --expiry {{ sas_expire }} --output=json
+    register: sasurl
+  
+  - name: set the sas URLS
+    set_fact:
+      openshift_azure_sas_url: "{{ bloburl.stdout|from_json + '?' + sasurl.stdout|from_json }}"
+  
+  - name: set the image tags
+    set_fact:
+      image_tags: "{{ (imagetags.stdout | from_json)[0] }}"
```
Comment (jim-minter) [inline playbooks/azure/openshift-cluster/create_offer.yml]: @maxamillion this is an example of the `{{ (foo.stdout | from_json)... }}` syntax that is bugging me ;)

File: playbooks/azure/openshift-cluster/templates/offer.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    microsoft-azure-marketplace.termsOfUse: "<p>This offering requires the use of an available OpenShift Container Platform subscription eligible for use in Azure.  If you don't currently have a subscription, you can obtain one <a href=\"http://www.redhat.com/contact/sales\">directly from Red Hat</a>.</p>\n  \n<p>Before using the OpenShift Container Platform subscription in Azure, you must enroll in the <a href=\"http://www.redhat.com/en/technologies/cloud-computing/cloud-access\">Red Hat Cloud Access program</a>.</p>\n\n<p>In order to deploy OpenShift Container Platform (Cloud Access) in Azure, you will need to provide your Red Hat Customer Portal login credentials so that the Azure instances can be registered with Red Hat Subscription Manager properly, and you must have enough OpenShift Container Platform entitlements to cover the chosen configuration.</p>\n \n<p>Your use of the Red Hat Product is governed by your Subscription Agreement with Red Hat. The standard Red Hat Subscription Agreement is located at <a href=\"http://www.redhat.com/licenses\">http://www.redhat.com/licenses</a></p>"
```
Comment (jim-minter) [inline playbooks/azure/openshift-cluster/templates/offer.yml.j2]: please replace with dummy text

**Comment by jim-minter** [comment]
tox errors:
```
playbooks/azure/openshift-cluster/create_offer.yml:
  [2m8:1[0m       [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m13:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m18:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m28:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m32:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m36:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m42:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m47:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m54:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m58:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m74:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m
  [2m78:1[0m      [31merror[0m    trailing spaces  [2m(trailing-spaces)[0m

playbooks/azure/openshift-cluster/group_vars/all/image_publish.yml:
  [2m4:12[0m      [31merror[0m    too many spaces after colon  [2m(colons)[0m
  [2m5:18[0m      [31merror[0m    too many spaces after colon  [2m(colons)[0m
  [2m18:12[0m     [31merror[0m    too many spaces after colon  [2m(colons)[0m
  [2m19:18[0m     [31merror[0m    too many spaces after colon  [2m(colons)[0m
```

**Comment by maxamillion** [comment]
@kwoodson those azure modules have a copyright mark from 2016, where did they originate?

**Comment by jim-minter** [comment]
@sdodson are you ok for this to be merged? footprint outside of the azure playbook is 2 new files in roles/lib_utils/library.

**Comment by sdodson** [comment]
Yeah that's fine.
/lgtm

**Comment by vrutkovs** [comment]
bot, retest this please

**Comment by kwoodson** [comment]
@jim-minter this is certainly unrelated to gcp.  I vote to merge.

**Pull request merged**